resolwe_bio/processes/variant_calling/snpeff.py

```diff
@@ -0,0 +1,236 @@
+"""Variant annotation with SnpEff."""
+
+
+from resolwe.process import (
+    BooleanField,
+    Cmd,
+    DataField,
+    FileField,
+    FileHtmlField,
+    GroupField,
+    ListField,
+    Persistence,
+    Process,
+    SchedulingClass,
+    StringField,
+)
+
+
+class SnpEff(Process):
+    """Annotate variants with SnpEff.
+
+    SnpEff is a variant annotation and effect prediction tool.
+    It annotates and predicts the effects of genetic variants
+    (such as amino acid changes).
+
+    This process also allows filtering of variants with ``SnpSift
+    filter`` command and extracting specific fields from the VCF
+    file with ``SnpSift extractFields`` command.
+    """
+
+    slug = "snpeff"
+    name = "snpEff (General variant annotation)"
+    process_type = "data:variants:vcf:snpeff"
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {"image": "public.ecr.aws/genialis/resolwebio/snpeff:2.1.1"},
+        },
+        "resources": {
+            "cores": 2,
+            "memory": 16384,
+        },
+    }
+    entity = {
+        "type": "sample",
+    }
+    category = "Other"
+    data_name = "Annotated variants (SnpEff)"
+    version = "1.0.0"
+    scheduling_class = SchedulingClass.BATCH
+    persistence = Persistence.CACHED
+
+    class Input:
+        """Input fields to SnpEff process."""
+
+        variants = DataField(
+            data_type="variants:vcf",
+            label="Variants (VCF)",
+        )
+        database = StringField(
+            label="snpEff database",
+            default="GRCh37.75",
+            choices=[("GRCh37.75", "GRCh37.75"), ("GRCh38.99", "GRCh38.99")],
+        )
+        dbsnp = DataField(
+            data_type="variants:vcf",
+            label="Known variants",
+            description="List of known variants for annotation.",
+            required=False,
+        )
+        filtering_options = StringField(
+            label="Filtering expressions",
+            description="Filter VCF file using arbitraty expressions."
+            "Examples of filtering expressions: '(ANN[*].GENE = 'PSD3')' "
+            "or '( REF = 'A' )' or "
+            "'(countHom() > 3) | (( exists INDEL ) & (QUAL >= 20)) | (QUAL >= 30 )'."
+            "For more information checkout the official documentation of [SnpSift]"
+            "(https://pcingola.github.io/SnpEff/ss_filter/)",
+            required=False,
+        )
+        sets = ListField(
```

Comment:
You could use the existing `data:geneset:*` object types. The gene_id / gene_name notation will need to be matched to fit the snpEff requrements, but otherwise it should work.